src/Hydrators/PropertyHydrator.php

```diff
@@ -15,11 +17,11 @@ public function __construct(\PDO $database)
         $this->database = $database;
     }
 
-    public function getAllBasicProperties()
+    public function getAllBasicProperties(): IPropertyCollection
     {
         $query = $this->database->query("SELECT `Address1`, `Town`, `propertyType`, `Status`, `id`,  `Image` FROM `properties`;");
```

Comment:
Field name don't match table.
also I believe that you should link the status and types tables to the properties table and use the respective field names.